### PR TITLE
Plant Suggestion UI

### PR DESCRIPTION
**Pull request by Bushuo**
Ref: #435 
Implements the right toolbar of the plant layer, which is a list of suggestions.
Also implements the basic functionality of the plant layer.

## Basics

- [x] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildserver is happy.
- [x] The PR is rebased with current master.

## Checklist

- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

File: frontend/src/features/map_planning/layers/plant/components/PlantSuggestions.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  available: MOCK_PLANTS,
+  diversity: MOCK_PLANTS,
+  favorites: MOCK_PLANTS,
+  recent: MOCK_PLANTS,
+};
+
+export function PlantSuggestions() {
+  return (
+    <div className="flex flex-col gap-4 p-2">
+      <PlantSuggestionList header={'Available Seeds'}>
```
Comment (markus2330) [inline frontend/src/features/map_planning/layers/plant/components/PlantSuggestions.tsx:50]: See #435, for first version only this should be enough.

**Review by markus2330 — COMMENTED** [review]
Great to see something here! :heart: :fireworks:

**Comment by Bushuo** [comment]
@markus2330 Do you agree with how the interaction works at the moment?
Please take a look at it.

**Comment by markus2330** [comment]
I like it (@yvonnemarkl didn't have a look yet). I assume that the left toolbar should cancel the planting and it is only a bug that this doesn't happen?

**Comment by Bushuo** [comment]
> I assume that the left toolbar should cancel the planting and it is only a bug that this doesn't happen?
>
Yes, using another tool, like from the left toolbar, should cancel any other previous tool.
I did not look into all the other interactions yet, because I wanted to quickly get the first preview.

**Comment by markus2330** [comment]
Yvonne likes it, too.

Some things we noticed:

- the planting is not centered and depends on:
  - where you've moved the canvas (middle mouse)
  - how much you've zoomed
- which plant gets planted is correctly shown bottom middle but the highlighting on the right toolbar might get lost
- when clicking on the currently selected plant in the right toolbar, the planting should stop, too
- when switching layer, the planting should be canceled, too
- connected/offline does not work: you can do plantings despite being offline (no connection to backend)

Please create issues for what you don't plan to do within this PR.

**Comment by Bushuo** [comment]
I should probably add some components to storybook. That is still missing and I will add them here.
I was thinking about the following:
- [ ] PlantSuggestionList
- [ ] PlantListItem

File: frontend/src/features/map_planning/layers/plant/PlantsLayer.tsx [hunk 2]
```diff
@@ -1,12 +1,199 @@
...
+      const position = e.target.getRelativePointerPosition();
+      if (!position) {
+        return;
+      }
+
+      executeAction(
+        new CreatePlantAction({
+          id: uuid.v4(),
+          plantId: selectedPlant.id,
+          // TODO: get the selectedLayerId from the store
```
Comment (markus2330) [inline frontend/src/features/map_planning/layers/plant/PlantsLayer.tsx:40]: better create left-over issues

**Review by markus2330 — APPROVED** [review]
Excellent job, we both love it! Works great on several browsers concurrently! Adding of seeds works *perfectly*.

There are some obvious problems in UI (which already existed before but are now more immanent), see #464

**Pull request merged**